daemon/info.go

```diff
@@ -111,3 +112,12 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
 
 	return v, nil
 }
+
+// HTTP_PROXY, HTTPS_RPOXY and NO_PROXY are case-insensitive.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func getProxyEnv(key string) string {
+	if os.Getenv(key) == "" {
+		return os.Getenv(strings.ToUpper(key))
```

Comment:
The way net/http behaves is that the uppercase variants are tried first.
Also, this function relies on the input being lowercase. If you call `getProxyEnv("HTTP_PROXY")`, the lowercase variant won't be tried.
Basically, should explicitly call `ToUpper` on the first try, and `ToLower` on the second.